torchvision/prototype/datasets/_builtin/sbd.py

```diff
@@ -0,0 +1,179 @@
+        )
+        segmentation = torch.as_tensor(raw_segmentation) if decode_segmentation else None
+
+        return boundaries, segmentation
+
+    def _collate_and_decode_sample(
+        self,
+        data: Tuple[Tuple[Any, Tuple[str, io.IOBase]], Tuple[str, io.IOBase]],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> Dict[str, Any]:
+        split_and_image_data, ann_data = data
+        _, image_data = split_and_image_data
+        image_path, image_buffer = image_data
+        ann_path, ann_buffer = ann_data
+
+        image = decoder(image_buffer) if decoder else image_buffer
+
+        if config.boundaries or config.segmentation:
+            boundaries, segmentation = self._decode_ann(
+                read_mat(ann_buffer), decode_boundaries=config.boundaries, decode_segmentation=config.segmentation
+            )
+        else:
+            boundaries = segmentation = None
+
+        return dict(
+            image_path=image_path,
+            image=image,
+            ann_path=ann_path,
+            boundaries=boundaries,
+            segmentation=segmentation,
+        )
+
+    def _make_datapipe(
+        self,
+        resource_dps: List[IterDataPipe],
+        *,
+        config: DatasetConfig,
+        decoder: Optional[Callable[[io.IOBase], torch.Tensor]],
+    ) -> IterDataPipe[Dict[str, Any]]:
+        archive_dp, extra_split_dp = resource_dps
+
+        archive_dp = resource_dps[0]
+        archive_dp = TarArchiveReader(archive_dp)
+        split_dp, images_dp, anns_dp = Demultiplexer(
+            archive_dp,
+            3,
+            self._classify_archive,  # type: ignore[arg-type]
+            buffer_size=INFINITE_BUFFER_SIZE,
+            drop_none=True,
+        )
+
+        if config.split == "train_noval":
+            split_dp = extra_split_dp
+        # TODO: replace the .map() call with decode=True in LineReader when
+        #  https://github.com/pytorch/data/issues/28 is resolved
+        split_dp = LineReader(split_dp).map(bytes.decode, input_col=1)
+        split_dp = Shuffler(split_dp)
+
+        dp = split_dp
+        for level, data_dp in enumerate((images_dp, anns_dp)):
+            dp = KeyZipper(
+                dp,
+                data_dp,
+                key_fn=getitem(*[0] * level, 1),
+                ref_key_fn=path_accessor("stem"),
+                buffer_size=INFINITE_BUFFER_SIZE,
+            )
+        return Mapper(dp, self._collate_and_decode_sample, fn_kwargs=dict(config=config, decoder=decoder))
+
+    def generate_categories_file(self, root: Union[str, pathlib.Path]) -> None:
+        dp = self.resources(self.default_config)[0].to_datapipe(pathlib.Path(root) / self.name)
+        dp = TarArchiveReader(dp)
+        dp = Filter(dp, path_comparator("name", "category_names.m"))
+        dp = LineReader(dp)
+        dp = Mapper(dp, bytes.decode, input_col=1)
+        lines = tuple(zip(*iter(dp)))[1]
```

Comment:
This is seems also a part `Mapper` can improve by only keeping the result of function and dropping all other columns.

Reply:
Maybe achieved by adding the `__getitem__` for `IterDataPipe` to select columns.

Reply:
The main goal is to somehow align the API with TorchArrow DataFrame since DataFrame normally can select column by `df['x']`.